File: core/src/com/alexpi/awesometanks/entities/blocks/Mine.java
package com.alexpi.awesometanks.entities.blocks;

import com.alexpi.awesometanks.entities.DamageListener;
import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.physics.box2d.CircleShape;
import com.badlogic.gdx.physics.box2d.World;

/**
 * Created by Alex on 20/02/2016.
 */
public class Mine extends Block {
    public Mine(DamageListener listener, AssetManager manager, World world, int posX, int posY) {
        super(manager,"sprites/mine.png",  world,new CircleShape(),150, posX, posY, .5f, false, listener);
    }
}

File: core/src/com/alexpi/awesometanks/entities/blocks/Mine.kt
package com.alexpi.awesometanks.entities.blocks

import com.alexpi.awesometanks.entities.components.body.BodyShape
import com.alexpi.awesometanks.entities.components.body.FixtureFilter
import com.alexpi.awesometanks.screens.game.stage.GameContext
import com.badlogic.gdx.math.Vector2

/**
 * Created by Alex on 20/02/2016.
 */
class Mine(gameContext: GameContext, pos: Vector2) : HealthBlock(gameContext, "sprites/mine.png", BodyShape.Circular(.25f), pos, 150f, true, false, FixtureFilter.BLOCK, false) {
    private val explosionManager = gameContext.getExplosionManager()
    private fun explode() {
        val mineX = bodyComponent.body.position.x
        val mineY = bodyComponent.body.position.y
        explosionManager.createLandMineExplosion(mineX, mineY)
    }


    override fun remove(): Boolean {
        explode()
        return super.remove()
    }
}